src/app/facilities/facilities.component.ts

```diff
@@ -0,0 +1,78 @@
+import {AfterViewInit, Component, ElementRef} from '@angular/core';
+import {FacilitiesContents} from "./utilities/faciilities-descriptions";
+import {fadeInOut, fadeInStill} from "../shared/animations";
+import {FacilityType} from "./utilities/FacilityContent.model";
+
+@Component({
+  selector: 'app-facilities',
+  templateUrl: './facilities.component.html',
+  styleUrls: ['./facilities.component.scss'],
+  animations: [fadeInOut, fadeInStill]
+})
+export class FacilitiesComponent implements AfterViewInit {
+    protected readonly FacilitiesContents = FacilitiesContents;
+    //TODO make proper type here
+    protected contentToDisplay: FacilityType = 'beach';
+    //TODO put this property in another variable
+    private observer = new IntersectionObserver((entries) => {
+        entries.forEach((entry) => {
+            if (entry.isIntersecting) {
+                entry.target.classList.add('visible');
+            }
+        });
+    }, { threshold: 0.2 });
+    constructor(private elementRef: ElementRef) {}
+
+    ngAfterViewInit() {
+        document.querySelectorAll('app-facility-descripton-mobile').forEach((element) => {
+            this.observer.observe(element);
+        });
+
+
+    }
+
+    setSelection(idName: string){
+        document.querySelector("#hint-arrow")?.classList.add("hide");
+        if(this.contentToDisplay === idName as FacilityType)
+            return;
+
+        document.querySelector(".selected")?.classList.remove('selected');
+        document.querySelector("#"+idName)?.classList.add('selected');
+
+        this.scrollToSection('facilitiesSection');
+
+        this.contentToDisplay = idName as FacilityType;
+
+    }
+    scrollToSection(sectionId: string) {
+        const targetElement = this.elementRef.nativeElement.querySelector('#'+ sectionId);
+        if (targetElement) {
+            targetElement.scrollIntoView({ behavior: 'smooth' });
+        }
+    }
```

Comment:
Generalnie takie bezpośredne operacje na DOM'ie w angularze nie są zalecane i mimo że można tak robić to nie jest to najlepsza praktyka, strzelam że wiesz o tym, tylko czasu brakowało, ale będzie trzeba to poprawić

Reply:
Nie nie wiedziałem o tym, że nie są zalecane. Chodzi Ci dokładnie o operacje na classList ?
Jeśli tak, to specjalnie tak napisałem, ponieważ robiąc to narzędziami, które udostępnia Angular kod byłby niesamowicie zawiły( chyba, że nie wpadłem na pomysł jak to zrobić w elegancki sposób)

Reply:
Tutaj masz bardziej to wytłumaczone
https://angular.io/guide/security
https://stackoverflow.com/questions/53451918/why-not-to-interact-with-the-dom-directly-in-angular